Checker.cpp

```diff
@@ -2,21 +2,48 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
+bool checktemp (float temp)
+{
+    if(temp < 0 || temp > 45) {
+        cout << "Temperature out of range!\n";
+        return false;
+    }
+
+    return true;
+}
+
+bool checksoc (float soc)
+{
+    if(soc < 20 || soc > 80){
+        cout << "State of Charge out of range!\n";
```

Comment:
Encapsulate cout statement to another function ! let checksoc remain as pure function